backend/mem/fileSystem.go

```diff
@@ -0,0 +1,176 @@
+package mem
+
+import (
+	"path"
+	"time"
+
+	"github.com/c2fo/vfs/v5"
+	"github.com/c2fo/vfs/v5/backend"
+	"github.com/c2fo/vfs/v5/utils"
+)
+
+//Scheme defines the filesystem type.
+const Scheme = "mem"
+const name = "In-Memory Filesystem"
+
+// FileSystem implements vfs.Filesystem for the mem filesystem.
+type obj struct {
+	isFile bool
+	i      interface{}
+}
+type objMap map[string]*obj
+type FileSystem struct {
+	fsMap map[string]objMap
+}
+
+// FileSystem will return a retrier provided via options, or a no-op if none is provided.
+func (fs *FileSystem) Retry() vfs.Retry {
+	return vfs.DefaultRetryer()
+}
+
+/*
+NewFile function returns the mem implementation of vfs.File.
+Since this is inside "fileSystem" we assume that the caller knows that the CWD is the root.
```

Comment:
"fileSystem" should be capitalized if referring to the type, two words, if referring to the underlying concept.